### PR TITLE
added scraping post draft

### PR DESCRIPTION
**Pull request by adipasquale**

### CODE REVIEW

File: _posts/2019-12-10-scraping-tips.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+The website tries to identify robots and prevent them from accessing the data, without impacting actual users browsing the website.
+The robots are thus designed to mimick as closely as possible actual users' behaviour, so as to become stealth.
+
+Here is a list of recommendations, more or less obvious, but always useful to remind of.
+
+
+## Keep It Simple, Stupid
+
+Nowadays, websites frontends are quit complex and often use powerful frameworks like React.
```
Comment (AntoineAugusti) [inline _posts/2019-12-10-scraping-tips.md]: Quite complex

File: _posts/2019-12-10-scraping-tips.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+Do not forget that browsing websites comes down to HTTP requests being sent to servers.
+
+There are two approaches to scraping:
+
+1. reproduce individual HTTP requests one by one
```
Comment (AntoineAugusti) [inline _posts/2019-12-10-scraping-tips.md]: Don't know if you need to expand on this. Hitting a URL is sometimes not enough. You need to send specific headers or a user agent, whatever

File: _posts/2019-12-10-scraping-tips.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+2. mimick an actual browser and user clicking around
+
+💁🏽‍♀️ As often as you can, try to use the first method. It is usually much more reliable and efficient.
+
+However, the two approaches are complimentary.
+Sometimes the second one is necessary, if the interactions between the JS and the HTTP requests are too entangled.
+
+## Sniff out APIs
+
+Look for public APIs that return JSON instead of HTML that you have to parse.
```
Comment (AntoineAugusti) [inline _posts/2019-12-10-scraping-tips.md]: Maybe include a GIF to show what it looks like?

File: _posts/2019-12-10-scraping-tips.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+*Pro Tip*: copied curl commands can be lengthy.
+I often copy them in my IDE, search and replace all `-H` with `-H \n` (with regex mode activated) to get one line per header.
+
+⚙️ [Python to curl](https://curl.trillworks.com/) is a tool that converts curl commands to Python `requests` call.
+
+## Best practices for scrapers
+
+- Fetch first then parse. This way you won't have to refetch if there is a bug in your parser.
+- You can also use caching locally when developping to bypass this problem
+- Distribute the workload in successive steps. eg. for a cooking website, first retrieve all recipes urls and then fetch them instead of directly following every nested link.
```
Comment (AntoineAugusti) [inline _posts/2019-12-10-scraping-tips.md]: URLs (casing)

File: _posts/2019-12-10-scraping-tips.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+Use and abuse the "Copy as curl" feature in your network tab, to paste it in a terminal and see if it's reproductible.
+What I often do is try and find the smallest set of headers and body params that still succeds by bissecting down.
+
+*Pro Tip*: copied curl commands can be lengthy.
+I often copy them in my IDE, search and replace all `-H` with `-H \n` (with regex mode activated) to get one line per header.
+
+⚙️ [Python to curl](https://curl.trillworks.com/) is a tool that converts curl commands to Python `requests` call.
+
+## Best practices for scrapers
```
Comment (AntoineAugusti) [inline _posts/2019-12-10-scraping-tips.md]: I'd say store a copy of the response / page if it fails. It's too bad to trash something that could be reprocessed later and that you way never be able to get again

File: _posts/2019-12-10-scraping-tips.md (hunk 2 above)
Comment (AntoineAugusti) [reply _posts/2019-12-10-scraping-tips.md]: Okay it's mentionned later so it may not be needed

File: _posts/2019-12-10-scraping-tips.md (hunk 5 above)
Comment (adipasquale) [reply _posts/2019-12-10-scraping-tips.md]: yes c'est ce qu'on essayait de dire dans le 'fetch first and then parse' j'ai étayé pour que ce soit plus clair

File: _posts/2019-12-10-scraping-tips.md (hunk 3 above)
Comment (adipasquale) [reply _posts/2019-12-10-scraping-tips.md]: good idea, done!

File: _posts/2019-12-10-scraping-tips.md (hunk 1 above)
Comment (adipasquale) [reply _posts/2019-12-10-scraping-tips.md]: simplified it, thanks!

**Pull request merged**